src/nvim/window.c

```diff
@@ -1131,6 +1131,7 @@ int win_split(int size, int flags)
 
   // When the ":tab" modifier was used open a new tab page instead.
   if (may_open_tabpage() == OK) {
+    apply_autocmds(EVENT_TABNEWENTERED, NULL, NULL, false, curbuf);
```

Comment:
why isn't this in `win_new_tabpage` ? oh, I see a similar pattern here: https://github.com/neovim/neovim/blob/133e2990efa44cadf1410a452a08843dbd8ca86e/src/nvim/ex_docmd.c#L5027-L5030

everything is so spread out :( but this could instead be done in `may_open_tabpage`, at least (otherwise what's the point of having `may_open_tabpage` ?):

```diff
diff --git a/src/nvim/window.c b/src/nvim/window.c
index 16bb7f5df794..4c8c3e1049c1 100644
--- a/src/nvim/window.c
+++ b/src/nvim/window.c
@@ -4261,7 +4261,11 @@ int may_open_tabpage(void)
 
   cmdmod.cmod_tab = 0;         // reset it to avoid doing it twice
   postponed_split_tab = 0;
-  return win_new_tabpage(n, NULL);
+  int ok = win_new_tabpage(n, NULL);
+  if (ok == OK) {
+    apply_autocmds(EVENT_TABNEWENTERED, NULL, NULL, false, curbuf);
+  }
+  return ok;
 }
 
 // Create up to "maxcount" tabpages with empty windows.
```

Reply:
yes. plan to check may_open_tabpage carefully when I wake up :)